app/controllers/categories_controller.rb

```diff
@@ -1,10 +1,54 @@
 class CategoriesController < ApplicationController
   def show
-    @category = Category.find_by(title: params[:category])
+    @category = Category.find_by(slug: params[:slug])
     render file: "public/404" unless @category
   end
 
   def index
     @categories = Category.all
   end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      flash[:success] = "Category created succesfully."
+      redirect_to dashboard_path
+    else
+      flash.now[:danger] = "Title can not be blank."
```

Comment:
we could write this message this way instead:
flash.now[:danger] = @category.errors.full_message.join(", ")